api.planx.uk/send/uniform.js

```diff
@@ -224,47 +237,68 @@ export async function createZip({
     columns: ["question", "responses", "metadata"],
     header: true,
   }).pipe(csvFile);
-  await new Promise((resolve, reject) => {
-    csvStream.on("error", reject);
-    csvStream.on("finish", resolve);
-  });
+  await resolveStream(csvStream);
   zip.addLocalFile(csvPath);
   deleteFile(csvPath);
 
-  // build the XML file from a string, write it locally, add it to the zip
-  //   must be named "proposal.xml" to be processed by Uniform
-  const xmlPath = "proposal.xml";
+  // add a XML uniform submission file to zip
+  const xmlPath = "proposal.xml"; //  must be named "proposal.xml" to be processed by Uniform
   const xmlFile = fs.createWriteStream(xmlPath);
-  const xmlStream = str(stringXml.trim()).pipe(xmlFile);
-  await new Promise((resolve, reject) => {
-    xmlStream.on("error", reject);
-    xmlStream.on("finish", resolve);
-  });
+  const xmlStream = str(xml.trim()).pipe(xmlFile);
+  await resolveStream(xmlStream);
   zip.addLocalFile(xmlPath);
   deleteFile(xmlPath);
 
-  // build an HTML Document Viewer
-  const docViewPath = path.join(tmpDir, "review.html");
-  const docViewFile = fs.createWriteStream(docViewPath);
-  const docViewStream = generateDocumentReviewStream({
-    csv,
-    files,
-    geojson,
-  }).pipe(docViewFile);
-  await new Promise((resolve, reject) => {
-    docViewStream.on("error", reject);
-    docViewStream.on("finish", resolve);
-  });
-  zip.addLocalFile(docViewPath);
-  deleteFile(docViewPath);
+  // generate and add an HTML overview document for the submission to zip
+  const overviewPath = path.join(tmpDir, "Overview.htm");
+  const overviewFile = fs.createWriteStream(overviewPath);
+  const overviewStream = generateHTMLOverviewStream(csv).pipe(overviewFile);
+  await resolveStream(overviewStream);
+  zip.addLocalFile(overviewPath);
+  deleteFile(overviewPath);
 
-  // build an optional GeoJSON file for validators
+  // add an optional GeoJSON file to zip
+  const geojson = passport.data["property.boundary.site"];
```

Comment:
DrawBoundary is a unique Editor component in that it has default passport variables - technically it can still be changed by content editors, but I agree it's fair to assume it'll always work here. See here for reference https://github.com/theopensystemslab/planx-new/blob/main/editor.planx.uk/src/%40planx/components/DrawBoundary/model.ts#L30